src/commands/server/delete.ts

```diff
@@ -58,6 +59,28 @@ export default class Delete extends Command {
           }
         }
       },
+      {
+        title: 'Delete the CR eclipse-che of type checlusters.org.eclipse.che',
+        task: async (_ctx: any, task: any) => {
+          if (await kh.crdExist('checlusters.org.eclipse.che') &&
+            await kh.cheClusterExist('eclipse-che', flags.chenamespace)) {
+            await kh.deleteCheCluster('eclipse-che', flags.chenamespace)
+            await cli.wait(2000) //wait a couple of secs for the finalizers to be executed
+            task.title = await `${task.title}...OK`
+          } else {
+            task.title = await `${task.title}...CR not found`
+          }
+        }
+      },
+      {
+        title: 'Delete CRD checlusters.org.eclipse.che',
+        task: async (_ctx: any, task: any) => {
+          if (await kh.crdExist('checlusters.org.eclipse.che')) {
+            await kh.deleteCrd('checlusters.org.eclipse.che')
+          }
+          task.title = await `${task.title}...OK`
+        }
+      },
       {
         title: 'Delete all deployments',
```

Comment:
So I assume this will stop the `che-operator` POD along with the various PODs of the Che server, Keycloak server and Postgres db, as well as of the workspaces that might have been started from this Che server.

Would it be safer to do this in successive steps:
1. Trigger deployment deletion for workspaces created from the Che server and wait for the effective deletion
2. Trigger deployment deletion for Che components (Che server, Keycloak and Postgress) and wait for the effective deletion 
3. Finally trigger deletion of the `che-operator` deployment ?

Reply:
Yes I think that it can be an improvement of the delete command. Probably for another PR. But I am wondering if we really need to do this: this is not a stop command, this is a delete/reset command were we reset all the resources and we are not really interested about the integrity but about doing a good cleanup.

Reply:
Sure, I understand. I just a bit worried about the fact that deleting the `che-operator` too early would let things in a state that would prevent correct cleaning of the rest.
Do we wait until the effective deletion of the `CheCluster` CR before triggering the deletion of deployments ?

Reply:
We wait for 2s after the deletion of the `CheCluster`. Tthis can be improved: we should probably wait for the deletion of this resource and other resources as well (che-server etc...). But I would rather fix that in a separated PR.

Reply:
OK to do it in a next PR